store/tikv/region_cache.go

```diff
@@ -385,6 +385,9 @@ func (c *RegionCache) loadStoreAddr(bo *Backoffer, id uint64) (string, error) {
 	for {
 		store, err := c.pdClient.GetStore(bo.ctx, id)
 		if err != nil {
+			if errors.Cause(err) == goctx.Canceled {
+				return "", errors.Trace(err)
+			}
 			err = errors.Errorf("loadStore from PD failed, id: %d, err: %v", id, err)
 			if err = bo.Backoff(boPDRPC, err); err != nil {
```

Comment:
Sounds reasonable, but I don't want to do it in this PR.
1. Backoffer already consider the sleep interval, and how many times to retry. 
    Add what kind of error is retryable make it more complex.
2. Change backoffer to skip context cancel error just help, but can't fix the problem, so I prefer this PR just do one fix.